src/main/java/net/rithms/util/RiotApiUtil.java

```diff
@@ -20,11 +20,15 @@
 import java.util.Iterator;
 import java.util.Map;
 import java.util.Objects;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 
 import net.rithms.riot.api.RiotApi;
 
 public final class RiotApiUtil {
 
+	private static final Pattern SUMMONER_VALIDATOR = Pattern.compile("^[0-9\\p{L} _.]+$");
```

Comment:
I know this regular expression was taken from the original Riot post mentioning it, but I wonder how accurate it actually is in comparison to their in-game validation. 

For example, character length might be something we also want to consider.

Reply:
The getting started [guide](https://developer.riotgames.com/getting-started.html) only mentions the regular expression in regards to validation. But the [Summoner Name FAQ](https://support.riotgames.com/hc/en-us/articles/201752814-Summoner-Name-FAQ#h1q4) specifies 3-16 characters for summoner names. I can't find anything indicating that the length would differ between regions or languages, so I assume the name length is universal. 
Also, does this mean that 3 spaces is a valid summoner name?

I can add the length check of 3-16 characters as well if you think it would be a good idea.

Reply:
It is possible for names to be longer than 16 characters (e.g. `IS1e8f1cd8ad5033482c781` on NA). It is not possible for a player to pick a name this long, but Riot can set an account's name to something longer than 16 characters.

Reply:
Technically Riot can also set a summoner name to something using characters that a user couldn't use himself.